From 5b14988845b591f6fa2fc1e032618fe882827f4a Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <fsumsal@redhat.com>
Date: Mon, 14 Oct 2019 16:22:51 +0200
Subject: [PATCH] travis: drop SCL remains

The `bash -ic` wrapper existed solely to make SCL work as expected

Resolves: #1761519
---
 ci/travis-centos-rhel8.sh | 16 +++-------------
 1 file changed, 3 insertions(+), 13 deletions(-)

diff --git a/ci/travis-centos-rhel8.sh b/ci/travis-centos-rhel8.sh
index ade44a0413..da131c726b 100755
--- a/ci/travis-centos-rhel8.sh
+++ b/ci/travis-centos-rhel8.sh
@@ -113,22 +113,12 @@ for phase in "${PHASES[@]}"; do
             $DOCKER_EXEC ninja -C build test
             ;;
         RUN_ASAN|RUN_CLANG_ASAN)
-            # Note to my future frustrated self: docker exec runs the given command
-            # as sh -c 'command' - which means both .bash_profile and .bashrc will
-            # be ignored. That's because .bash_profile is sourced for LOGIN shells (i.e.
-            # sh -l), whereas .bashrc is sourced for NON-LOGIN INTERACTIVE shells
-            # (i.e. sh -i).
-            # As the default docker exec command lacks either of those options,
-            # we need to use a wrapper command which runs the wanted command
-            # under an explicit bash -i, so the SCL source above works properly.
-            docker exec -it $CONT_NAME bash -ic 'gcc --version'
-
             if [[ "$phase" = "RUN_CLANG_ASAN" ]]; then
                 ENV_VARS="-e CC=clang -e CXX=clang++"
                 MESON_ARGS="-Db_lundef=false" # See https://github.com/mesonbuild/meson/issues/764
             fi
-            docker exec $ENV_VARS -it $CONT_NAME bash -ic "meson build --werror -Dtests=unsafe -Db_sanitize=address,undefined $MESON_ARGS ${CONFIGURE_OPTS[@]}"
-            docker exec -it $CONT_NAME bash -ic 'ninja -v -C build'
+            docker exec $ENV_VARS -it $CONT_NAME meson build --werror -Dtests=unsafe -Db_sanitize=address,undefined $MESON_ARGS ${CONFIGURE_OPTS[@]}
+            docker exec -it $CONT_NAME ninja -v -C build
 
             # Never remove halt_on_error from UBSAN_OPTIONS. See https://github.com/systemd/systemd/commit/2614d83aa06592aedb.
             travis_wait docker exec --interactive=false \
@@ -136,7 +126,7 @@ for phase in "${PHASES[@]}"; do
                 -e ASAN_OPTIONS=strict_string_checks=1:detect_stack_use_after_return=1:check_initialization_order=1:strict_init_order=1 \
                 -e "TRAVIS=$TRAVIS" \
                 -t $CONT_NAME \
-                bash -ic 'meson test --timeout-multiplier=3 -C ./build/ --print-errorlogs'
+                meson test --timeout-multiplier=3 -C ./build/ --print-errorlogs
             ;;
         CLEANUP)
             info "Cleanup phase"
